frontend/src/components/FilterBar.tsx

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from 'react';
+import * as Autosuggest from 'react-autosuggest';
+import Chip from '@material-ui/core/Chip';
+import ChipInput, { ChipRendererArgs } from 'material-ui-chip-input';
+import MenuItem from '@material-ui/core/MenuItem';
+import Paper from '@material-ui/core/Paper';
+import { ApiFilter } from '../apis/filter';
+import {
+  SuggestionsFetchRequestedParams,
+  SuggestionHighlightedParams,
+  SuggestionSelectedEventData
+} from 'react-autosuggest';
+import { classes } from 'typestyle';
+// TODO: match and parse required editing the node modules to import correctly.
+// Perhaps we should consider using allowSyntheticDefaultImports, although we could pretty easily
+// implement these functions ourselves if need be.
+// tslint:disable-next-line:no-var-requires
+const match = require('autosuggest-highlight/match');
+// tslint:disable-next-line:no-var-requires
+const parse = require('autosuggest-highlight/parse');
+
+interface FilterChip {
+  filterValue: number | string;
```

Comment:
Can this be a number? We shouldn't care I think, we can just treat all values as strings.